cli/src/pcluster/validators/cluster_validators.py

```diff
@@ -57,7 +58,7 @@
 class ClusterNameValidator(Validator):
     """Cluster name validator."""
 
-    def _validate(self, name):
+    def _validate(self, name, config):
```

Comment:
I don't like too much to reuse the ClusterNameValidator.

`ClusterNameValidator` is called in the `_validate_and_parse_config` outside the logic of resource validators and so it cannot be suppressed with `validator_suppressors`

```python
    validation_failures = ClusterNameValidator().execute(name=self.name)
    validation_failures += config.validate(validator_suppressors)
```

Then I don't like the fact we're passing the config to the ClusterNameValidator. 
We have to find some alternatives to move ClusterNameValidator into the BaseClusterConfig validation to be suppressed or find a way to pass the stack name to the HostedZoneValidator.

Let's talk about it.

Reply:
After we discuss about it, seems there no better solution since it is hard to pass stack name into HostedZoneValidator. Currently leave it as it is

Reply:
Rebased on the patch#2960 and move the validation to DomainNameValidator